src/aead/aes.rs

```diff
@@ -263,29 +258,29 @@ impl Key {
                     }
                     ctr32_encrypt_blocks!(
                         GFp_bsaes_ctr32_encrypt_blocks,
-                        &mut in_out[..(bsaes_in_out_len + in_prefix_len)],
-                        in_prefix_len,
+                        &mut in_out[src.clone()][bsaes_in_out_len..],
+                        src.clone(),
                         &bsaes_key,
                         ctr
```

Comment:
I may be mistaken but this looks incorrect - it will encrypt a suffix of the input rather than a prefix?